api/v1/institutions/PKPInstitutionHandler.inc.php

```diff
@@ -0,0 +1,224 @@
+        }
+
+        $collector->filterByContextIds([$this->getRequest()->getContext()->getId()]);
+
+        HookRegistry::call('API::institutions::params', [$collector, $slimRequest]);
+
+        $institutions = Repo::institution()->getMany($collector);
+
+        return $response->withJson([
+            'itemsMax' => $institutions->count(),
+            'items' => Repo::institution()->getSchemaMap()->summarizeMany($institutions),
+        ], 200);
+    }
+
+    /**
+     * Add an institution
+     *
+     * @throws \Exception For sending a request to the API endpoint of a particular context.
+     */
+    public function add(SlimHttpRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $request = $this->getRequest();
+
+        if (!$request->getContext()) {
+            throw new \Exception('You can not add an institution without sending a request to the API endpoint of a particular context.');
+        }
+
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_INSTITUTION, $slimRequest->getParsedBody());
+        $params['contextId'] = $request->getContext()->getId();
+        // Convert IP ranges string to array
+        if (!empty($params['ipRanges'])) {
+            $ipRanges = explode(PHP_EOL, trim($params['ipRanges']));
+            $params['ipRanges'] = $ipRanges;
+        }
+
+        $primaryLocale = $request->getContext()->getPrimaryLocale();
+        $allowedLocales = $request->getContext()->getSupportedFormLocales();
+        $errors = Repo::institution()->validate(null, $params, $allowedLocales, $primaryLocale);
+        if (!empty($errors)) {
+            return $response->withStatus(400)->withJson($errors);
+        }
+
+        $institution = Repo::institution()->newDataObject($params);
+        $id = Repo::institution()->add($institution);
+        $institution = Repo::institution()->get($id);
+        return $response->withJson(Repo::institution()->getSchemaMap()->map($institution), 200);
+    }
+
+    /**
+     * Edit an institution
+     */
+    public function edit(SlimHttpRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $request = $this->getRequest();
+        $context = $request->getContext();
+
+        if (!Repo::institution()->existsByContextId((int) $args['institutionId'], $context->getId())) {
+            return $response->withStatus(404)->withJsonError('api.institutions.404.institutionNotFound');
+        }
+
+        $institution = Repo::institution()->get((int) $args['institutionId']);
+
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_INSTITUTION, $slimRequest->getParsedBody());
+        $params['id'] = $institution->getId();
+        $params['contextId'] = $context->getId();
+        // Convert IP ranges string to array
+        if (!empty($params['ipRanges'])) {
+            $ipRanges = explode(PHP_EOL, trim($params['ipRanges']));
+            $params['ipRanges'] = $ipRanges;
+        }
```

Comment:
I will leave it as protected in this class. APIHandler already has `paramToArray($value): array` that assumes the params are comma separated. This is maybe rather an exception that they are separated by EOL, because of the form...